### PR TITLE
Documentation Link Checker

### PR DESCRIPTION
**Pull request by paulhazen**
This PR contains the script I created to identify broken links in the documentation, and a workflow for GitHub that triggers whenever a markdown file has changed. Configured properly, it can be used to prevent merging broken links into `stable`, `development`, and/or `release` branches (whichever we would prefer).

### CODE REVIEW

File: tools/scripts/check_links.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        if not is_external_link(link):
+            abs_link = compute_abs_link(md_dir, link)
+    else:
+        if is_external_link(link):
+            try:
+                response = session.head(link, allow_redirects=True, timeout=10, headers=headers)
+                if response.status_code != 403 and (response.status_code >= 400 or response.status_code == 405):
+                    response = session.get(link, allow_redirects=True, timeout=10, headers=headers)
+                    if response.status_code >= 400 and response.status_code != 403:
+                        result = response.status_code
```
Comment (paulhazen) [inline tools/scripts/check_links.py]: This needs more documentation to explain why `403` is excluded from consideration.

Comment (paulhazen) [reply tools/scripts/check_links.py]: This has been updated to include documentation and to be clearer.

File: .github/workflows/check-markdown-links.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Check Markdown Links
+
+on:
+  push:
+    branches: ["feat/documentation-link-checker"]
```
Comment (andrew-hirata-playeveryware) [inline .github/workflows/check-markdown-links.yml:5]: This requires another change later to set it to the correct branch, right?

**Review by andrew-hirata-playeveryware — APPROVED** [review]

File: .github/workflows/check-markdown-links.yml (hunk 2 above)
Comment (paulhazen) [reply .github/workflows/check-markdown-links.yml:5]: That is correct. I figure that determining _where_ we want this to be enabled would be a conversation for another PR.

**Review by arthur740212 — APPROVED** [review]

**Pull request merged**